segmentation/include/pcl/segmentation/impl/unary_classifier.hpp

```diff
@@ -309,8 +307,6 @@ pcl::UnaryClassifier<PointT>::queryFeatureDistances (std::vector<pcl::PointCloud
   {
     // Query point  
     flann::Matrix<float> p = flann::Matrix<float>(new float[n_col], 1, n_col);
-    memcpy (&p.ptr ()[0], (*query_features)[i].histogram, p.cols * p.rows * sizeof (float));
```

Comment:
This line should not be removed (I think).

Reply:
None of the other `flann::Matrix` variables are initialized; does `knnSearch` return garbage if it's not?

Reply:
:man_shrugging: I don't know. Will take a deeper look and get back to you

Reply:
This definitely has to be initialized since its an input for `knnSearch` (`indices` and `distances` are output).
`std::copy((*query_features)[i].histogram, (*query_features)[i].histogram + n_col, p.ptr())` should work (`1*n_col` is the same as `p.cols*p.rows`)